File: apps/backend/src/services/auth_service.py
from abc import ABC, abstractmethod
from pathlib import Path


class AuthService(ABC):
    """
    认证服务接口，定义了认证相关的抽象方法
    """

    @abstractmethod
    async def cookie_auth_douyin(self, account_file: Path) -> bool:
        """抖音cookie认证"""
        pass

    @abstractmethod
    async def cookie_auth_tencent(self, account_file: Path) -> bool:
        """腾讯视频号cookie认证"""
        pass

    @abstractmethod
    async def cookie_auth_ks(self, account_file: Path) -> bool:
        """快手cookie认证"""
        pass

    @abstractmethod
    async def cookie_auth_xhs(self, account_file: Path) -> bool:
        """小红书cookie认证"""
        pass

    @abstractmethod
    async def check_cookie(self, platform_type: int, file_path: str) -> bool:
        """检查指定平台的cookie有效性"""
        pass


class DefaultAuthService(AuthService):
    """
    默认认证服务实现
    """

    async def cookie_auth_douyin(self, account_file: Path) -> bool:
        """抖音cookie认证"""
        from src.services.cookie_service import get_cookie_service

        return await get_cookie_service().cookie_auth_douyin(account_file)

    async def cookie_auth_tencent(self, account_file: Path) -> bool:
        """腾讯视频号cookie认证"""
        from src.services.cookie_service import get_cookie_service

        return await get_cookie_service().cookie_auth_tencent(account_file)

    async def cookie_auth_ks(self, account_file: Path) -> bool:
        """快手cookie认证"""
        from src.services.cookie_service import get_cookie_service

        return await get_cookie_service().cookie_auth_ks(account_file)

    async def cookie_auth_xhs(self, account_file: Path) -> bool:
        """小红书cookie认证"""
        from src.services.cookie_service import get_cookie_service

        return await get_cookie_service().cookie_auth_xhs(account_file)

    async def check_cookie(self, platform_type: int, file_path: str) -> bool:
        """检查指定平台的cookie有效性"""
        from src.services.cookie_service import get_cookie_service

        return await get_cookie_service().check_cookie(platform_type, file_path)


# 全局默认认证服务实例
default_auth_service = DefaultAuthService()

File: apps/backend-node/src/services/auth-service.ts
/**
 * Authentication service for omni-post backend (Node.js).
 * Mirrors: apps/backend/src/services/auth_service.py
 */

import { CookieService, getCookieService } from './cookie-service.js';

/**
 * AuthService interface — delegates to CookieService
 */
export interface AuthService {
    cookieAuthDouyin(accountFile: string): Promise<boolean>;
    cookieAuthWxChannels(accountFile: string): Promise<boolean>;
    cookieAuthKs(accountFile: string): Promise<boolean>;
    cookieAuthXhs(accountFile: string): Promise<boolean>;
    cookieAuthBilibili(accountFile: string): Promise<boolean>;
}

/**
 * Default implementation — delegates to CookieService
 */
export class DefaultAuthService implements AuthService {
    private cookieService: CookieService;

    constructor(cookieService?: CookieService) {
        this.cookieService = cookieService ?? getCookieService();
    }

    async cookieAuthDouyin(accountFile: string): Promise<boolean> {
        return this.cookieService.cookieAuthDouyin(accountFile);
    }

    async cookieAuthWxChannels(accountFile: string): Promise<boolean> {
        return this.cookieService.cookieAuthWxChannels(accountFile);
    }

    async cookieAuthKs(accountFile: string): Promise<boolean> {
        return this.cookieService.cookieAuthKs(accountFile);
    }

    async cookieAuthXhs(accountFile: string): Promise<boolean> {
        return this.cookieService.cookieAuthXhs(accountFile);
    }

    async cookieAuthBilibili(accountFile: string): Promise<boolean> {
        return this.cookieService.cookieAuthBilibili(accountFile);
    }
}

// Singleton
let _authService: AuthService | null = null;

export function getAuthService(): AuthService {
    if (!_authService) {
        _authService = new DefaultAuthService();
    }
    return _authService;
}

/**
 * Set auth service (for testing / dependency injection)
 */
export function setAuthService(service: AuthService): void {
    _authService = service;
}
